### ISSUE TITLE
Bug: False warning for property mismatch when using resolve_target_entities

### ISSUE THREAD

**Issue by CoalaJoe** [phpstan/phpstan-doctrine#121]
I get a lot of errors like these:
```
Property App\Entity\Example::$example  mapping mismatch: database can contain  App\Entity\OtherExample|null but property expects App\Contract\OtherExampleInterface.
```

In the doctrine configuration under `doctrine.orm`, you can configure mappings of Interfaces to entities. I basically have something like this:

```yaml
doctrine:
  # ...
  orm:
    # ...
    resolve_target_entities:
      App\Contract\OtherExampleInterface: App\Entity\OtherExample
```

**Comment by ondrejmirtes** [comment]
Please show your property example with annotation.

Also there's this https://github.com/phpstan/phpstan-doctrine/issues/98 so I suspect that:

1) This might be a duplicate.
2) The `|null` part suggests your mapping is actually wrong.

**Comment by CoalaJoe** [comment]
You are actually right! Thank you, i somehow did not realize the `|null`-part.
Declaring the JoinColumn with `nullable=true` resolved some errors.

The remaining warnings are mostly the same as in the issue you've mentioned. I am closing this as a duplicate.
